pkg/commands/plan/plan_init.go

```diff
@@ -0,0 +1,243 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package plan provide the Terraform planning functionality for Guardian.
+package plan
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/abcxyz/guardian/pkg/flags"
+	"github.com/abcxyz/guardian/pkg/git"
+	"github.com/abcxyz/guardian/pkg/github"
+	"github.com/abcxyz/guardian/pkg/terraform"
+	"github.com/abcxyz/guardian/pkg/util"
+	"github.com/abcxyz/pkg/cli"
+	"github.com/abcxyz/pkg/logging"
+	gh "github.com/google/go-github/v53/github"
+)
+
+var _ cli.Command = (*PlanInitCommand)(nil)
+
+type PlanInitCommand struct {
+	cli.BaseCommand
+
+	workingDir  string
+	entrypoints []string
+
+	flags.GitHubFlags
+	flags.RetryFlags
+
+	flagPullRequestNumber      int
+	flagDestRef                string
+	flagSourceRef              string
+	flagDirectories            []string
+	flagDeleteOutdatedComments bool
+	flagJSON                   bool
+
+	gitClient    git.Git
+	githubClient github.GitHub
+
+	// testFlagSetOpts is only used for testing.
+	testFlagSetOpts []cli.Option
+}
+
+func (c *PlanInitCommand) Desc() string {
+	return `Run the Terraform plan for a directory.`
+}
+
+func (c *PlanInitCommand) Help() string {
+	return `
+Usage: {{ COMMAND }} [options]
+
+  Initialize Guardian for running the Terraform plan process.
+`
+}
+
+func (c *PlanInitCommand) Flags() *cli.FlagSet {
+	set := cli.NewFlagSet(c.testFlagSetOpts...)
+
+	c.GitHubFlags.AddFlags(set)
+	c.RetryFlags.AddFlags(set)
+
+	f := set.NewSection("Command options")
+
+	f.IntVar(&cli.IntVar{
+		Name:    "pull-request-number",
+		Target:  &c.flagPullRequestNumber,
+		Example: "100",
+		Usage:   "The GitHub pull request number associated with this plan run.",
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:    "dest-ref",
+		Target:  &c.flagDestRef,
+		Example: "ref-name",
+		Usage:   "The destination GitHub ref name for finding file changes.",
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:    "source-ref",
+		Target:  &c.flagSourceRef,
+		Example: "ref-name",
+		Usage:   "The source GitHub ref name for finding file changes.",
+	})
+
+	f.StringSliceVar(&cli.StringSliceVar{
+		Name:    "directories",
+		Target:  &c.flagDirectories,
+		Example: "dir1,dir2,dir3",
+		Usage:   "The subset of directories to target for Terraform planning.",
+	})
+
+	f.BoolVar(&cli.BoolVar{
+		Name:    "delete-outdated-comments",
+		Target:  &c.flagDeleteOutdatedComments,
+		Default: true,
```

Comment:
This is more of a Go-style thing. Usually booleans default to "false" (since the zero value is false), so it'd be more appropriate to have:

```shell
-keep-outdated-comments
```

and default it to false.